### ISSUE TITLE
`Esc` in editor pane propagates to console

### ISSUE THREAD

**Issue by david-cortes** [rstudio/rstudio#9839]
**Steps to reproduce:**
* Run some command that would respond to interrupt signals and that would take a long time to run (for example try `install.packages("mlpack")` on a linux computer).
* While the command is running, switch the focus to the editor.
* Press `Esc`.

**Expected behavior:** the key that was pressed should only have a possible effect on the pane that is currently selected (the editor).

**Actual behavior:** it sends a stop signal to the console.

**Setup:** RStudio 2021.09.0 Build 340, running on debian bookworm (sid).

**Comment by mikebessuille** [comment]
Need to determine if this is a regression in Ghost Orchid.

**Comment by hadley** [comment]
Are you sure this is a regression? I think the current behaviour is expected as there are some commands (like `Esc`) that  affect the source pane regardless of which pane has the focus.

**Comment by david-cortes** [comment]
But `Esc` is also used in the editor pane, for example to cancel a search or a function signature pop-up. One can press it accidentally trying to accomplish something else and ending up canceling the running command.

**Comment by hadley** [comment]
@david-cortes right, I think that could be an argument for not introducing this behaviour in the first place. But you are proposing a change in behaviour, which I think a lot of people would find annoying because they (like me) are used to pressing `Esc` to cancel a running computation anywhere.

**Comment by valerie-rstudio** [comment]
Confirmed that this same behavior is in Juliet Rose (1.4.1717-3). This is not a regression in Ghost Orchid.
